From 47c1b757eed7aaef1c6a34db01626f9ead8fb5f6 Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@nxp.com>
Date: Wed, 18 Jul 2018 13:04:57 +0300
Subject: [PATCH 4215/5242] MLK-18955-3: ASoC: fsl_spdif: restore arate type

commit  7365b8f353e1eb1497457f77674257ee66b3ae06 from
https://source.codeaurora.org/external/imx/linux-imx.git

Changing "arate" type to u64 makes 32-bit kernel build to fail,
so restore the previous type.

Fixes: bb7d18078220 ("MLK-18955-2: ASoC: fsl_spdif: fix sysclk_df type")

Signed-off-by: Viorel Suman <viorel.suman@nxp.com>
Reviewed-by: Leonard Crestez <leonard.crestez@nxp.com>
(cherry picked from commit 5250b586a5975d27efe73a5b4ca8d8e9c27cc6ab)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_spdif.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sound/soc/fsl/fsl_spdif.c b/sound/soc/fsl/fsl_spdif.c
index 462316c..7b72b21 100644
--- a/sound/soc/fsl/fsl_spdif.c
+++ b/sound/soc/fsl/fsl_spdif.c
@@ -1212,7 +1212,8 @@ static u32 fsl_spdif_txclk_caldiv(struct fsl_spdif_priv *spdif_priv,
 {
 	static const u32 rate[] = { 32000, 44100, 48000, 96000, 192000 };
 	bool is_sysclk = clk_is_match(clk, spdif_priv->sysclk);
-	u64 rate_actual, sub, arate;
+	u64 rate_actual, sub;
+	u32 arate;
 	u16 sysclk_dfmin, sysclk_dfmax, sysclk_df;
 	u8 txclk_df;
 
-- 
1.7.9.5

